prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -33,13 +36,21 @@
 GENERAL_AGENT_TAG = "general_agent"
 
 
-class DeployableMicrochainAgent(DeployableAgent):
+class DeployableMicrochainAgentAbstract(DeployableAgent, metaclass=abc.ABCMeta):
     model = SupportedModel.gpt_4o
-    n_iterations = 50
-    load_historical_prompt: bool = False
-    system_prompt_choice: SystemPromptChoice = SystemPromptChoice.TRADING_AGENT
-    identifier: AgentIdentifier = AgentIdentifier.MICROCHAIN_AGENT_OMEN
-    description: str
+    max_iterations: int | None = 50
+    import_actions_from_memory = 0
+    identifier: AgentIdentifier
+    functions_config: FunctionsConfig
+
+    @classmethod
+    def get_description(cls) -> str:
+        return f"Microchain-based {cls.__name__}."
+
+    @classmethod
+    @abc.abstractmethod
+    def get_initial_system_prompt(cls) -> str:
```

Comment:
Same thing for system prompt, but this has some other implications as well in this PR, as will be seen in a while.